### DISCUSSION TITLE
Possible for bash-like tab behaviour, to fills prompt with candidate common prefix

### DISCUSSION THREAD

**Discussion by Lenbok** [minad/corfu#395] [Q&A] [answered]
I'm quite used to tab completion in bash, where pressing tab will pre-fill the prompt area with the common prefix of completion candidates (then you generally type the next char of the desired candidate to further refine, followed by tab).

Is this interaction mode possible in corfu?

Here's what I'm currently using:

```
(use-package corfu
  :init
  (global-corfu-mode)
  :bind (:map corfu-map ("RET" . nil))
  :config
  (setq corfu-auto t
        corfu-left-margin-width 2
        corfu-right-margin-width 2
        corfu-on-exact-match nil
        corfu-preselect 'directory
        corfu-quit-no-match 'separator))
```

Currently tab grabs the first matching candidate.

**Comment by minad** [comment, marked as answer]
~~~ elisp
(setq corfu-preselect 'prompt
      completion-styles '(basic orderless))
~~~

**Comment by Lenbok** [comment]
I don't use `orderless` - is that a requirement? (It does not work for me with my current `completion-styles` of `'(prescient basic)`). For example:
![image](https://github.com/minad/corfu/assets/282098/9def69fe-30a0-4100-9b6f-9344c24b5a06)
pressing tab now does nothing, whereas it should fill in "aliz". If this works for you then I must have some other kind of misconfiguration.

**Reply by minad** [reply]
Try `(basic prescient)`, since `basic` must be listed first in order to enable prefix expansion.

**Reply by Lenbok** [reply]
Ahhh, that does the trick, thank you! I guess I'll take a look at `prescient` and/or `corfu-prescient` to see if it can put prescient last in the styles list or whether I'll need to explicitly set that..

**Reply by Lenbok** [reply]
Seems like this is sufficient:
```
(use-package corfu-prescient
  :after prescient corfu
  :config
  (setq corfu-prescient-completion-styles '(basic prescient)
        corfu-prescient-completion-category-overrides '((file (styles basic prescient))))
  (corfu-prescient-mode 1))
```
